Lsunae/ElevenStreetSample/app/src/main/java/com/exam/elevenstreet/view/product/ProductActivity.kt

```diff
@@ -0,0 +1,101 @@
+package com.exam.elevenstreet.view.product
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.exam.elevenstreet.network.api.ElevenStreetApi
+import com.exam.elevenstreet.RetrofitInstance
+import com.exam.elevenstreet.view.product.adapter.ProductAdapter
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.activity_main.recycler_view_product
+import java.net.URL
+import android.os.AsyncTask
+import com.exam.elevenstreet.R
+import com.exam.elevenstreet.data.parser.ProductXmlPullParserHandler
+import com.exam.elevenstreet.data.repository.ProductRepository
+import com.exam.elevenstreet.data.source.local.ProductLocalDataSource
+import com.exam.elevenstreet.data.source.remote.ProductRemoteDataSource
+import com.exam.elevenstreet.network.model.ProductResponse
+
+
+
+class ProductActivity : AppCompatActivity() {
+
+    private var elevenStreetApi: ElevenStreetApi? = null
+    private lateinit var productAdapter: ProductAdapter
+    private lateinit var keyWord: String
+    private var pageNum = 1
+
+    private val productRepository by lazy { ProductRepository(ProductRemoteDataSource(), ProductLocalDataSource()) }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(com.exam.elevenstreet.R.layout.activity_main)
+
+        val productList = ProductLocalDataSource().productData()
+
+
+        productAdapter = ProductAdapter(productList.toMutableList())
+
+        elevenStreetApi =
+            RetrofitInstance.getInstance<ElevenStreetApi>("https://openapi.11st.co.kr/openapi/")
+
+        setupView()
+
+        btn_search.setOnClickListener {
+
+            productRepository.getProductList()
+
+            keyWord = "${edit_search.text}"
+            pageNum = 1
+
+            val productList = ProductTask().execute(keyWord, pageNum.toString()).get()
```

Comment:
```suggestion
            val productList = ProductTask().execute(keyWord, pageNum).get()
```
이렇게 작성되는게 하는게 더 좋지 않을까 생각합니다.